pyrtls.pyi

```diff
@@ -0,0 +1,407 @@
+"""
+# pyrtls: rustls-based modern TLS for Python
+
+[Latest version](https://pypi.org/project/pyrtls) |
+[Documentation](https://pyrtls.readthedocs.io) |
+[CI](https://github.com/djc/pyrtls/actions?query=workflow%3ACI+branch%3Amain)
+
+pyrtls provides bindings to [rustls](https://github.com/rustls/rustls), a modern Rust-based TLS implementation with an API that is
+intended to be easy to use to replace the `ssl` module (but not entirely compatible with it).
+
+In addition to being memory-safe, the library is designed to be more secure by default. As such,
+it does not implement older protocol versions, cipher suites with known security problems, and
+some problematic features of the TLS protocol. For more details, review the [manual](https://docs.rs/rustls/latest/rustls/manual/index.html).
+
+## Warning!
+
+This project is just getting started. While rustls is mature, the Python bindings
+are pretty new and not yet feature-complete.
+"""
+
+from collections.abc import Iterable
+from socket import socket
+
+# lib.rs
+
+class TrustAnchor:
+    def __new__(
+        cls,
+        subject: bytes,
+        subject_public_key_info: bytes,
+        name_constraints: bytes | None,
+    ) -> TrustAnchor: ...
+
+class IoState:
+    def tls_bytes_to_write(self) -> int:
+        """
+        How many bytes could be written by `Connection.write_tls_into()` if called right now.
+        A non-zero value implies that `Connection.wants_write()` would yield `True`.
+        """
+
+    def plaintext_bytes_to_read(self) -> int:
+        """How many plaintext bytes are currently buffered in the connection."""
+
+    def peer_has_closed(self) -> bool:
+        """
+        `True` if the peer has sent us a `close_notify` alert. This is the TLS mechanism to
+        securely half-close a TLS connection, and signifies that the peer will not send any
+        further data on this connection.
+        """
+
+class TLSError: ...
```

Comment:
I feel like the `TLSError` class is incomplete, but I can't figure out what to add. Maybe documentation?

Reply:
I'm not sure what would be useful here. At least it's now being exported (since #52).

Reply:
I synced my fork 👍